File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.cpp
#include <string>
#include <vector>

using namespace std;

int FibDp(int n) // DP
{
	int* dp = new int[n + 1];

	dp[1] = 1;
	dp[2] = 1;
	for (int i = 3; i <= n; i++)
	{
		dp[i] = (dp[i - 1] + dp[i - 2]) % 1234567;
	}
	return dp[n];
}

int solution(int n) 
{
    int answer = FibDp(n);
    return answer;
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.cs
public class Solution 
{
    public int solution(int n) 
    {
        int[] Fibo = new int[100001];
        Fibo[0] = 0;
        Fibo[1] = 1;
        for(int i=2; i<=n; i++)
        {
            Fibo[i] = (Fibo[i-2] + Fibo[i-1]) % 1234567;
        }
        return Fibo[n];
    }
}